server/sonar-server/src/test/java/org/sonar/server/test/ws/ListActionTest.java

```diff
@@ -192,6 +192,59 @@ public void list_based_on_main_file_and_line_number() throws Exception {
     request.execute().assertJson(getClass(), "list-main-file.json");
   }
 
+  @Test
+  public void list_based_on_source_file_key_and_line_number() throws Exception {
+    String sourceFileUuid = "MAIN-FILE-UUID";
+    String sourceFileKey = "MAIN-FILE-KEY";
+    userSessionRule.addProjectUuidPermissions(UserRole.CODEVIEWER, TestFile1.PROJECT_UUID);
+    dbClient.componentDao().insert(db.getSession(),
+      new ComponentDto()
```

Comment:
All these components creation could be factorised -> FILE_1, FILE_2, or something like that.